Makefile

```diff
@@ -3,15 +3,26 @@ CC = gcc
 CFLAGS  = -O0 -g -Wall -pthread
 LDFLAGS =
 INCLUDES = -I./mINI.c \
-	   -I/usr/include/gdk-pixbuf-2.0 \
-	   -I/usr/include/glib-2.0 \
-	   -I/usr/lib/glib-2.0/include \
-	   -I/usr/include/sysprof-4 \
-	   -I/usr/include/libpng16 \
-	   -I/usr/include/libmount \
-	   -I/usr/include/blkid
-LIBS = -lm -lpthread -lrt -lgpiod -lsystemd \
-       -lnotify -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0
+	   $(shell pkg-config --cflags \
+	     gdk-pixbuf-2.0 \
+	     glib-2.0 \
+	     libpng16 \
+	     mount \
+	     blkid \
+	     libgpiod \
```

Comment:
To be a bit pendandic, it would be perfect if the version is constrained with libgpiod, something like this:
````diff
	   $(shell pkg-config --cflags \
	     gdk-pixbuf-2.0 \
	     glib-2.0 \
	     libpng16 \
	     mount \
	     blkid \
-	     libgpiod \
+	     "libgpiod < 2.0" \
````

Of course, this does not change the fact that the current code will not compile with a rolling-release distro such as arch, but it informs that for the current code base the version will not match.
Perhaps it's a different story on Debian, sadly this doesn't work on my machine a 100%.

Reply:
It's possible to specify a max of min version like this:

```
pkg-config --max-version=1.6.3 libgpiod
```

That exits with a non-zero exit code if the version doesn't match. We could add a check for that in the Makefile and print an error message if the check fails. But I wonder if it would be better to document that in the readme instead.

I see that on Arch the .pc file for libgpiod is still called libgpiod.pc. Strange -- I would have expected it to have a version in the name of the file since there's an API incompatibility. In any case, to use a private version of libgpiod with pkg-config, you do this:

```
PKG_CONFIG_PATH=~/src/my-libgpiod/lib make
```

But you'll have to check me on the correct path to put there. It should be the directory where the libgpiod.pc file is, which gets generated during the build. I believe pkg-config searches /usr/local/include by default also, in case you `make install`ed your private libgpiod.

Reply:
Thank you for the valuable input.
I'll accept that the build fails for now, but I plan to port to the new API version as soon as I get around to do it.
It shouldn't be that hard, as IO access is somewhat abstracted.

After that, I'll try to make the IO chips and lines configurable for CM4 per define.